### PR TITLE
have sort_with_indices allocate. Add a couple convenience procs for using the indices result to sort other slices.

### PR DESCRIPTION
**Pull request by thePHTest**
I need to add some tests for the sort_by_indices procs. Also, would love any suggestions for better naming.

### CODE REVIEW

**Comment by Kelimion** [comment]
I just added tests for `slice.sort_with_indices`, Phil.

**Comment by thePHTest** [comment]
@Kelimion I should add tests for the new sort_by_indices procs that can sort a slice given the indices that were the result of sort_with_indices.

**Comment by Kelimion** [comment]
> @Kelimion I should add tests for the new sort_by_indices procs that can sort a slice given the indices that were the result of sort_with_indices.

Ah, I got you. Feel free to add them to `tests/core/slice`.

**Comment by Kelimion** [comment]
Let's keep this PR open until those new procedures are tested as well, and then merge it?

**Comment by thePHTest** [comment]
I added the tests. I could also add some more procs like `sort_by_cmp_with_indices` etc to make it exhaustive. Also, a sort_soa could be implemented fairly easily now. Or those could come later.

**Comment by thePHTest** [comment]
and if you have a better name for the procs than `sort_by_indices` and `sort_By_indices_overwrite` let me know and I'm happy to change them,

**Comment by thePHTest** [comment]
I suppose we could also add a `sort_by_indices_swap` that doesn't allocate or do the overwrite so that the caller can provide the slice to swap with.

File: tests/core/slice/test_core_slice.odin [hunk 1]
```diff
@@ -94,4 +91,69 @@ test_sort_with_indices :: proc(t: ^testing.T) {
...
+
+		// Sort
+		f_idx := slice.sort_with_indices(vals)
+		defer delete(f_idx)
+
+		// Verify sorted test values
+		rand.init(&r, seed)
+
+		{
+			indices := make([]int, test_size)
```
Comment (Kelimion) [inline tests/core/slice/test_core_slice.odin:129]: Looks like you're no longer cleaning up `indices := make([]int, test_size)` here or in the next block?

**Pull request merged**